tasks/README.md

```diff
@@ -144,6 +148,8 @@ In addition to the tasks parameters, the task runner may inject metaparameters p
 
 **_installdir**: Tasks with `files` specified in the metadata will be passed the `_installdir` metaparameter to provide the file path to the expected resources.(rev 3)
 
+**_target**: Connection information for connectin to the real target when the task is running on a proxy.
```

Comment:
Do we want to carry a schema here similar to task parameters?

Reply:
sending the schema down to the task next to the data doesn't seem to accomplish much. we could add a schema to task metadata but that would be painful to use since you'd have to copy it everywhere.  The correct solution for this IMO is to have a {{target-type}} setting in metadata. That would reference puppet device implementation that would own the schema. In the interest of getting this working I'm leaving that as a separate enhancement to the task spec.

Reply:
oh no, not sending the schema to the task, only having it in the task metadata (like the parameter schema).

When you're pointing at the target-type, I do agree that replicating the schema across tasks is "less than useful".

Reply:
I do think we need some concept of type in the inital iteration of this. We can't support `apply` or `facts` without knowing what provider to use. I think we can do that without supporting schema checking yet.